src/include/duckdb/execution/index/art/art_operator.hpp

```diff
@@ -62,6 +62,61 @@ class ARTOperator {
 		return nullptr;
 	}
 
+	//! NestedLookup returns true if rowid is in the nested ART starting at gate_node,
+	//! and returns false otherwise.
+	static bool NestedLookup(ART &art, const Node &gate_node, const ARTKey &rowid) {
```

Comment:
Maybe we can all this `LookupInLeaf` or similar? That way, it becomes a bit more universal, and the `gate_node` parameter (maybe just `node`) could also be `NType::LEAF_INLINED` - in which case we return early with via:

```cpp
if (node.GetType() == NType::LEAF_INLINED) {
return node.GetRowId() == rowid.GetRowId();
}
```

Then, we can also rewrite the `ART::Insert` debug path like this to ensure we inserted the expected row ID:
```cpp
// remove this
		if (leaf->GetGateStatus() == GateStatus::GATE_SET) {
			D_ASSERT(ARTOperator::NestedLookup(*this, *leaf, row_id_keys[i]));
		}
// instead, just
D_ASSERT(ARTOperator::LookupInLeaf(*this, *leaf, row_id_keys[i]));
```